### PR TITLE
Add hints for most of the intermediate challenges

### PR DESCRIPTION
**Pull request by piglei**
Add hints for most of the intermediate challenges.
Some hints might be not good enough, please feel free to modify them. @laike9m

### CODE REVIEW

**Comment by laike9m** [comment]
Awesome, thanks for the great work, will take a look today

File: challenges/intermediate-self/hints.md
Comment (laike9m) [inline challenges/intermediate-self/hints.md:1]: Do you mind add a soltuon for using `TypeVar` instead?

Comment (piglei) [reply challenges/intermediate-self/hints.md:1]: Sure, I will add one later.

Comment (piglei) [reply challenges/intermediate-self/hints.md:1]: Added.

**Review by laike9m — APPROVED** [review]

**Comment by laike9m** [comment]
Thanks again!

**Pull request merged**